### PR TITLE
Bak descriptions et al

### PR DESCRIPTION
**Pull request by bucaran**
Add proper descriptions, remove _should_, etc.

### CODE REVIEW

**Comment by bpinto** [comment]
Same as #313 ; we can only push it after #314 and #315.

**Comment by bucaran** [comment]
In that order?

**Comment by bpinto** [comment]
#315 would help solving #314. :)

**Comment by bucaran** [comment]
Alright. Will work on this first.

**Comment by bucaran** [comment]
Some of the changes that I will be pushing:

+ Remove spec.log and use `msg`.
+ Everything will be silent by default, but you can turn verbose optionally.
+ spec.eval will no longer evaluate multiple functions as there is no need. The main loop occurs in spec.run. 
+ An improved UI. Similar to mocha. With a failed tests report after the run.
+ Will add some more options to spec.eval such as `--title` to indicate the current function should be displayed as a header. These options should be semantic, as spec.run should not deal with any design issues such as colors, etc.
+ expect will no longer print anything. Instead it will notify the results back to spec.eval where output will be displayed accordingly and based on the --verbose flag.

Any comments?

File: plugins/bak/spec/bak.spec.fish [hunk 1]
```diff
@@ -18,24 +18,25 @@ function describe_bak -d 'Testing bak plugin'
     popd
   end
 
-  function it_checks_bak_filename_pattern_is_followed
+  function it_checks_bak_filename_pattern_is_followed \
+    -d "checks `bak` filename pattern is followed"
```
Comment (bucaran) [inline plugins/bak/spec/bak.spec.fish]: followed by what?

Comment (bpinto) [inline plugins/bak/spec/bak.spec.fish]: If we stick to the pattern. I guess. :P

Comment (bucaran) [inline plugins/bak/spec/bak.spec.fish]: :+1:

**Comment by bpinto** [comment]
> Everything will be silent by default, but you can turn verbose optionally.

Don't know what that means, but if by default it only returns status 0 or 1, I think this is not expected. I'd have a `--quiet` option for this instead.

I think by default it should output

```
......X.....X....

fish-spec/list.erase.fish erases an element failed:
Expected: X
Actual: y
```

**Comment by bucaran** [comment]
Edited.

**Comment by bpinto** [comment]
Sounds reasonable. :+1:

**Comment by bucaran** [comment]
:+1: OK

**Comment by bucaran** [comment]
I will call the PR `fish-spec-upgrade`. See you in a few hours.

**Comment by bucaran** [comment]
> The above implies that there must be only one describe, before_all after_all, etc., per test suite. This will probably be good enough for now, but we should improve this in the future.

**Comment by bpinto** [comment]
:+1:

**Comment by bucaran** [comment]
We should be able to have multiple describes if they are not nested. Would rather have multiple files for this, though.

**Comment by bucaran** [comment]
```fish
  function it_has_an_output_if_suite_is_blank
    set -l suite "
        import plugins/fish-spec

        function describe_blank_suite
        end

        spec.run
      "

    expect (run_nested_suite $suite) --to-equal 'No tests found.'
  end
```
Why? What is the advantage of this?

**Comment by bpinto** [comment]
Multiple describes if they are not nested is something we should support as we should support nested describes. I know nested describes is only coming in the future, but no reason to remove multiple describe blocks, if that is what you were thinking.

Specs without describe blocks should also output "No tests found."

**Pull request closed without merging**